File: Front/poke-tcg/src/app/models/Listing/card-listing.ts
import { ListingStatus } from './listing-status';
 
export interface CardListing {
  listingId: number;
  cardId: string;
  price: number;
  sellerId: number;
  buyerId: number | null;
  status: ListingStatus;
  createdDate: string;
  modifiedDate: string | null;
  description: string | null;
 
  cardName?: string;
  cardImage?: string;
  setId?: string;
}

File: Back/ApiTcg/Domain/Entities/CardListing.cs
using Domain.Enum;

namespace Domain.Entities;

public class CardListing
{
    public int ListingId { get; set; }
    public string CardId { get; set; } = "";
    public decimal Price { get; set; }
    public int SellerId { get; set; }
    public int? BuyerId { get; set; }
    public ListingStatus Status { get; set; } = ListingStatus.Active;
    public DateTime CreatedDate { get; set; }
    public DateTime? ModifiedDate { get; set; }
    public string? Description { get; set; }
}
